packages/blade/src/utils/useColorScheme/useColorScheme.ts

```diff
@@ -0,0 +1,31 @@
+import { useState, useCallback } from 'react';
+import { ColorSchemeNames, colorSchemeNames } from '../../tokens/theme';
+import getColorScheme from '../getColorScheme';
+
+export type UseColorScheme = {
+  colorScheme: Exclude<ColorSchemeNames, 'system'>;
+  setColorScheme: (colorScheme: ColorSchemeNames) => void;
+};
+
+const useColorScheme = (initialColorScheme: ColorSchemeNames = 'light'): UseColorScheme => {
+  // if colorscheme defined use that else fallback to 'light'
+  const [colorSchemeState, setColorSchemeState] = useState<Exclude<ColorSchemeNames, 'system'>>(
```

Comment:
Okay so we are supporting system. The places where you see `system` excluded are the places where we exclude them from being set in state. For example you say you want to switch to `system` theme but in the end all the APIs on web as well as native return you back `light` or `dark` so system is what you accept as an input and `light/dark` is what you set.

Reply:
and the places where you see the includes check it's basically when input. so
```jsx
const { colorScheme, setColorScheme } = useTheme()
setColorScheme('system')
```
so in the implementation of `setColorScheme` you see that includes check since it's a valid value as an input

Reply:
Okay so would it make sense to have

```
type ColorSchemeNames = 'light' | 'dark'

type ColorSchemeNamesInput = 'system' | ColorSchemeNames
```

<img width="773" alt="Screenshot 2021-04-20 at 2 48 23 PM" src="https://user-images.githubusercontent.com/24487274/115371341-840e9e80-a1e7-11eb-869a-adb84c6bf4c4.png">
<img width="794" alt="Screenshot 2021-04-20 at 2 48 14 PM" src="https://user-images.githubusercontent.com/24487274/115371350-8670f880-a1e7-11eb-9c4f-c5b6c4cf8c1a.png">